lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -20,7 +20,13 @@ def initialize(engine_config:)
         end
 
         def run(file)
-          process_file(file)
+          $stderr.puts("About to analyze #{file}. If analysis fails, consider excluding this file.")
+          skip, skip_reason = skip?(file)
```

Comment:
To avoid the multiple return values, I think I'd prefer this method return a reason or `nil`, so it's just

```rb
if (reason = skip?(file)`
```